### PR TITLE
Refactor `utils` and `functional`

### LINKED ISSUE #132: Utils refactoring
**Is your feature request related to a problem? Please describe.**
The project becomes more complicated. After each PR it includes new supplementary code. New structure seems to be needed.

**Describe the solution you'd like**

1. Make a `utils` folder
2. Put the `functional.py` here "as is"
3. Rename `utils.py` -> `utils/common.py`
3. Create a new file `utils/layers.py`
4. Add `L2Pool2d` from `perceptual.py` here (from #131)

**Describe alternatives you've considered**
Leave things "as is" or just move `L2Pool2d` to the `functional.py` as @denproc proposed in #131

### PR DESCRIPTION
**Pull request by zakajd**
Closes #132 

## Proposed Changes

  - Applied structure suggested by @denproc 
  - Included `__all__` in`__init__.py` files to avid PEP8 errors

### CODE REVIEW

File: piq/functional/base.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    r""" Similar to np.fft.ifftshift but applies to PyTorch Tensors
+    """
+    shift = [-(ax // 2) for ax in x.size()]
+    return torch.roll(x, shift, tuple(range(len(shift))))
+
+
+def get_meshgrid(size: Tuple[int, int]) -> torch.Tensor:
+    r"""
+    Args:
+        size: Shape of meshgrid to create
```
Comment (denproc) [inline piq/functional/base.py:15]: Wrong indentation here and following functions and files. 
```suggestion
        Args:
            size: Shape of meshgrid to create
```

File: piq/functional/colour_conversion.py [hunk 2]
```diff
@@ -1,77 +1,8 @@
-from typing import Tuple, Union
+r"""Color space convertion functions"""
```
Comment (denproc) [inline piq/functional/colour_conversion.py:1]:
```suggestion
r"""Colour space convertion functions"""
```

File: piq/functional/filters.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import torch
```
Comment (denproc) [inline piq/functional/filters.py:1]: Let's add short docstring to describe the file similar to colour space conversions.

File: piq/functional/base.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
+from typing import Tuple
+import torch
+
+
+def ifftshift(x: torch.Tensor):
+    r""" Similar to np.fft.ifftshift but applies to PyTorch Tensors
+    """
```
Comment (snk4tr) [inline piq/functional/base.py:7]:
```suggestion
    r""" Similar to np.fft.ifftshift but applies to PyTorch Tensors """
```

File: piq/functional/base.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
+from typing import Tuple
+import torch
+
+
+def ifftshift(x: torch.Tensor):
```
Comment (snk4tr) [inline piq/functional/base.py:5]:
```suggestion
def ifftshift(x: torch.Tensor) -> torch.Tensor:
```

File: piq/functional/base.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    r""" Similar to np.fft.ifftshift but applies to PyTorch Tensors
+    """
+    shift = [-(ax // 2) for ax in x.size()]
+    return torch.roll(x, shift, tuple(range(len(shift))))
+
+
+def get_meshgrid(size: Tuple[int, int]) -> torch.Tensor:
+    r"""
```
Comment (snk4tr) [inline piq/functional/base.py:13]: Could you give a one-two sentence description here?

File: piq/vsi.py [hunk 7]
```diff
@@ -10,7 +10,7 @@
 from torch.nn.modules.loss import _Loss
 from torch.nn.functional import avg_pool2d, interpolate, pad
 from piq.functional import ifftshift, gradient_map, scharr_filter, rgb2lmn, rgb2lab, similarity_map, get_meshgrid
-from .utils import _validate_input, _adjust_dimensions
+from piq.utils import _validate_input, _adjust_dimensions
```
Comment (snk4tr) [inline piq/vsi.py:13]: Yay! 🎉

**Comment by zakajd** [comment]
@snk4tr ready for merge

**Pull request merged**